src/foam/nanos/auth/CachingAuthService.java

```diff
@@ -66,7 +68,16 @@ public void eof() {
         public void reset(Detachable sub) {
           purgeCache(x);
         }
-      }, TRUE);
+      };
+
+      DAO userDAO       = (DAO) x.get("userDAO");
+      DAO groupDAO      = (DAO) x.get("groupDAO");
+      DAO groupPermissionJunctionDAO = (DAO) x.get("groupPermissionJunctionDAO");
+
+      groupDAO.listen(purgeSink, TRUE);
+      userDAO.listen(purgeSink, EQ(User.ID, user.getId()));
```

Comment:
Another thought about setting up these listeners. When a session is deleted/finished, are we removing/detaching the listeners? Wouldn't it be a memory leak if we aren't?

Reply:
Yes, we need to unsubscribe once the session is removed. How do we know when a session is stale?